MonoGame.Framework/Audio/Microphone.cs

```diff
@@ -128,7 +129,7 @@ public static ReadOnlyCollection<Microphone> All
         /// </summary>
         public static Microphone Default
         {
-            get { return _default; }
+            get { SoundEffect.Initialize(); return _default; }
```

Comment:
Write code like this in the static initializer instead of static property.

Reply:
Just so there is no confusion, ie:

```c#
static Microphone()
{
    SoundEffect.Initialize();
}
```

Reply:
You could also use static constructors in few other places.

Reply:
I think it's better to check if initialization was done properly on every call, rather than do it only once. It makes behavior more predictable. Static initializers complicate control flow because you can't be sure when they'll be called.

Reply:
They get called during first access to their respective class.

Reply:
Yes, I mean we don't control when it's called.

Reply:
The thing is, ```Default``` and ```All`` are the microphone entry points, but you have to initialize the sound API to populate the microphones, and the Microphone class is static by design...
Unless we change the design of the Microphone class, I don't think we can do anything else than having this in a static field. I'm not a fan of static classes because this kind of design problems.

Alternative proposition: not initializing the sound API at all in the Microphone class, and throw if All/Default are accessed while not initialized. Forcing users to either call ```SoundEffect.Initialize()``` or to use the auto initialization when loading content.

I've seen static classes initializing without obvious calls to them, so I'd feel more comfortable with not having any library loading happening inside static ctor/fields.

For the same reason, I don't quite like ```OpenAL.NativeLibrary``` being initialized statically. I think I'm going to change that too.

Reply:
We should not change the API by forcing explicit initialization. If we could have a fresh API, I'd definitely want to make it explicit though.

Reply:
How about just checking if Audio systems can be initialized on game run?

> For the same reason, I don't quite like OpenAL.NativeLibrary being initialized statically. I think I'm going to change that too.

Please don't, those are just wrappers. Think of them as `DllImport` statements :P

Reply:
Sure, but they do load the dll if the static ctor kicks in, which we want to avoid to not mess up with third party sound engine.

But if we can't get the sound api to initialize in a non static way for Microphone, there's no point in removing the static initialization from OpenAL anyway because it'll still be around within Microphone.

Reply:
I am more suspicious of Microphone than I am of OpenAL. I'd prefer to not add a static constructor to Microphone.

If we'd like to be 100% safe, I'd suggest to move static initialization out of OpenAL func loading and manually load them (to have a perfect control on the "when" and "how").